package NKikimrConfig;
option java_package = "ru.yandex.kikimr.proto";

message TMemoryControllerConfig {
    optional uint64 HardLimitBytes = 1;

    optional float SoftLimitPercent = 2 [default = 75];
    optional uint64 SoftLimitBytes = 3;

    optional float TargetUtilizationPercent = 4 [default = 50];
    optional uint64 TargetUtilizationBytes = 5;

    optional float ActivitiesLimitPercent = 6 [default = 30];
    optional uint64 ActivitiesLimitBytes = 7;

    optional float SharedCacheMinPercent = 100 [default = 20];
    optional uint64 SharedCacheMinBytes = 101;
    optional float SharedCacheMaxPercent = 102 [default = 50];
    optional uint64 SharedCacheMaxBytes = 103;

    optional float MemTableMinPercent = 110 [default = 1];
    optional uint64 MemTableMinBytes = 111;
    optional float MemTableMaxPercent = 112 [default = 3];
    optional uint64 MemTableMaxBytes = 113;

    optional float QueryExecutionLimitPercent = 120 [default = 20];
    optional uint64 QueryExecutionLimitBytes = 121;

    optional float ReadExecutionLimitPercent = 200 [default = 20]; // 1/3 - soft limit = ~6
    optional uint64 ReadExecutionLimitBytes = 201;
    optional float CompactionLimitPercent = 202 [default = 16]; // 2 2 6 6
    optional uint64 CompactionLimitBytes = 203;
    optional float ReadCacheLimitPercent = 204 [default = 4]; // 2 + 2
    optional uint64 ReadCacheLimitBytes = 205;
}
